dolphinscheduler-dao/src/main/resources/sql/dolphinscheduler_h2.sql

```diff
@@ -486,6 +486,8 @@ CREATE TABLE t_ds_task_definition
     delay_time              int(11) DEFAULT '0',
     task_group_id           int(11) DEFAULT NULL,
     task_group_priority     tinyint(4) DEFAULT '0',
+    cpu_quota               int(11) DEFAULT '-1' NOT NULL,
+    memory_max              int(11) DEFAULT '-1' NOT NULL,
```

Comment:
I think we should discuss about thoes column adding. Out table column is too many IMO, and this property is only use by some of the tasks not all of them, do we really need to add them into separate columns? should we add some column name `extra` or some name else, and put those two columns and others extra column only for some of the tasks in the further to keep our column as less column as possible?

Reply:
I've thought about it, too. But I can't find a field to put these two contents without affecting other functions....

Reply:
how about using single column as string and store and parse json

Reply:
We just split json in version 2.0+. I think we should not add more json columns on it...

Reply:
but currently our task param still in json now

Reply:
I am not sure about that, do you have time to take a look this discuss @caishunfeng @lenboo

Reply:
>how about using single column as string and store and parse json
Do you mean the `task_params`?
I think if the parameter is generic and matches all tasks, it can add a new column.

Reply:
> I think if the parameter is generic and matches all tasks, it can add a new column.
But it is not, it only for some of task types which may download resource file form resource center. and user have to use those resource file. for example users could use shell without resource file alone.

Reply:
> > I think if the parameter is generic and matches all tasks, it can add a new column.
> 
> But it is not, it only for some of task types which may download resource file form resource center. and user have to use those resource file. for example users can use or can not use shell with resource files.

ignore this comment, I finally get that we want to limit the linux resource in task, in this case I think we could add new column for it 🤣